airbyte-integrations/connectors/source-mongodb-internal-poc/src/main/java/io/airbyte/integrations/source/mongodb/internal/state/MongoDbStateManager.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.mongodb.internal.state;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.integrations.source.mongodb.internal.cdc.MongoDbCdcState;
+import io.airbyte.protocol.models.v0.AirbyteGlobalState;
+import io.airbyte.protocol.models.v0.AirbyteStateMessage;
+import io.airbyte.protocol.models.v0.AirbyteStreamNameNamespacePair;
+import io.airbyte.protocol.models.v0.AirbyteStreamState;
+import io.airbyte.protocol.models.v0.StreamDescriptor;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A state manager for MongoDB CDC syncs.
+ */
+public class MongoDbStateManager {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MongoDbStateManager.class);
+
+  /**
+   * The global CDC state.
+   */
+  private MongoDbCdcState cdcState;
+
+  /**
+   * Map of streams (name/namespace tuple) to the current stream state information stored in the
+   * state.
+   */
+  private final Map<AirbyteStreamNameNamespacePair, MongoDbStreamState> pairToStreamState = new HashMap<>();
+
+  /**
+   * Creates a new {@link MongoDbStateManager} primed with the provided initial state.
+   *
+   * @param initialState The initial state to be stored in the state manager.
+   * @return A new {@link MongoDbStateManager}
+   */
+  public static MongoDbStateManager createStateManager(final JsonNode initialState) {
+    final MongoDbStateManager stateManager = new MongoDbStateManager();
+
+    if (initialState != null) {
+      LOGGER.info("Initial state {}", initialState);
+      final List<AirbyteStateMessage> stateMessages = deserializeState(initialState);
+      if (!stateMessages.isEmpty()) {
+        if (stateMessages.size() == 1) {
+          final AirbyteStateMessage stateMessage = stateMessages.get(0);
+          stateManager.updateCdcState(Jsons.object(stateMessage.getGlobal().getSharedState(), MongoDbCdcState.class));
```

Comment:
Isn't the CDC state supposed to be the debezium state? How will it be de-serialised to `MongoDbCdcState`

Reply:
The `MongoDbCdcState` holds the data extracted from the Debezium offset map.  There is code to handle the translation to/from the offset map and `MongoDbCdcState` that is not in this PR, but will be in the PR that integrates all the bits together.